crates/opt8n/src/evm.rs

```diff
@@ -0,0 +1,37 @@
+use alloy::primitives::SignatureError;
+use anvil_core::eth::transaction::TypedTransaction;
+use revm::primitives::{OptimismFields, TxEnv, TxKind, U256};
+
+pub fn to_revm_tx_env(tx: TypedTransaction) -> Result<TxEnv, SignatureError> {
+    let transact_to = if let Some(to) = tx.to() {
+        TxKind::Call(to)
+    } else {
+        TxKind::Create
+    };
+
+    let from = tx.recover()?;
+    let op_fields: OptimismFields = match &tx {
+        TypedTransaction::Deposit(fields) => OptimismFields {
+            enveloped_tx: None,
+            source_hash: Some(fields.source_hash),
+            is_system_transaction: Some(fields.is_system_tx),
+            mint: Some(fields.mint.to()),
+        },
+        _ => OptimismFields::default(),
+    };
+    Ok(TxEnv {
+        gas_price: U256::from(tx.gas_price()),
+        gas_limit: tx.gas_limit() as u64,
+        value: U256::from(tx.value()),
+        nonce: Some(tx.nonce()),
+        caller: from,
+        transact_to,
+        data: tx.data().clone(),
+        chain_id: tx.chain_id(),
+        max_fee_per_blob_gas: tx.max_fee_per_blob_gas().map(U256::from),
+        access_list: tx.essentials().access_list.flattened(),
+        gas_priority_fee: tx.essentials().max_priority_fee_per_gas,
+        blob_hashes: tx.essentials().blob_versioned_hashes.unwrap_or_default(),
+        optimism: op_fields,
+    })
+}
```

Comment:
It looks like anvil core already has this method available for you [here](https://github.com/foundry-rs/foundry/blob/master/crates/anvil/core/src/eth/transaction/mod.rs#L444)

You would need to do:
```
let tx: TypedTransaction = ...; // Wherever your typed transaction comes from
let pending_tx: anvil_core::eth::transaction::PendingTransaction = PendingTransaction::from(tx)?;
let tx_env = pending_tx.to_revm_tx_env()?;
```